### DISCUSSION TITLE
Serializing FilterSet

### DISCUSSION THREAD

**Discussion by michaelastein** [carltongibson/django-filter#1561] [Q&A] [answered]
Hello, how can I serialize the Filterset in order to send it back to my angular Frontend?

views.py:
@api_view(['GET'])
def materialien_search(request):
    # GET list of materials with theses attributes
        filter = MaterialFilter(request.GET, queryset=Materialien.objects.all())
        materialien_serializer = MaterialienSerializer(filter, many=True)
        return JsonResponse(materialien_serializer.data, safe=False)
    
serializers.py:

class MaterialienSerializer(serializers.ModelSerializer):
    class Meta:
        model = Materialien
        fields = [field.name for field in Materialien._meta.get_fields()]

filter.py:
class MaterialFilter(django_filters.FilterSet):
    class Meta:
        model = Materialien
        fields = '__all__'

In the moment its not working because "TypeError: 'MaterialFilter' object is not iterable". I'd be very thankfull for any help.

**Comment by carltongibson** [comment, marked as answer]
The filter set is equivalent to the applied filters, which is just the GET QueryDict standardly.
